uportal-war/src/main/resources/properties/contexts/applicationContext.xml

```diff
@@ -34,16 +34,22 @@
     
     <context:annotation-config/>
     <context:component-scan base-package="org.jasig.portal">
-        <context:exclude-filter type="regex" expression="org\.jasig\.portal\.portlets\."/>
-        <context:exclude-filter type="regex" expression="org\.jasig\.portal\.security\.mvc\."/>
-        <context:exclude-filter type="regex" expression="org\.jasig\.portal\.json\.rendering\."/>
-        <context:exclude-filter type="regex" expression="org\.jasig\.portal\.layout\.dlm\.remoting\."/>
-        <context:exclude-filter type="regex" expression="org\.jasig\.portal\.security\.remoting\."/>
-        <context:exclude-filter type="regex" expression="org\.jasig\.portal\.rest\."/>
-        <context:exclude-filter type="regex" expression="org\.jasig\.portal\.redirect\."/>
-        <context:exclude-filter type="regex" expression="org\.jasig\.portal\.rendering\."/>
+        <context:exclude-filter type="regex" expression="org\.jasig\.portal\.portlets\..*"/>
+        <context:exclude-filter type="regex" expression="org\.jasig\.portal\.security\.mvc\..*"/>
+        <context:exclude-filter type="regex" expression="org\.jasig\.portal\.json\.rendering\..*"/>
+        <context:exclude-filter type="regex" expression="org\.jasig\.portal\.layout\.dlm\.remoting\..*"/>
+        <context:exclude-filter type="regex" expression="org\.jasig\.portal\.security\.remoting\..*"/>
+        <context:exclude-filter type="regex" expression="org\.jasig\.portal\.rest\..*"/>
+        <context:exclude-filter type="regex" expression="org\.jasig\.portal\.redirect\..*"/>
+        <context:exclude-filter type="regex" expression="org\.jasig\.portal\.rendering\..*"/>
     </context:component-scan>
-    
+
+    <!-- Required for TenantService -->
```

Comment:
Ha!  I see what's happening here, and it adds up... uPortal is in need of some "services-layer" rework.

In many cases the "services" we have grew organically out of administrative portlets we were building.  We cobbled together an API inside an o.j.p.portlets.* package (impl package for an admin portlet), and it fell into use by other areas of the portal.  The result is that now we can't "fix" the bug in <context:component-scan>.

I think we should look briefly at where we can set up proper services in short order (maybe nowhere), and provide a comment explaining why we're crossing these boundaries everywhere else.